GVFS/GVFS.Platform.Linux/LinuxFileSystemVirtualizer.cs

```diff
@@ -0,0 +1,719 @@
+                                {
+                                    activity.RelatedError(metadata, $"{nameof(this.virtualizationInstance.WriteFileContents)} failed, error: " + result.ToString("X") + "(" + result.ToString("G") + ")");
+                                    throw new GetFileStreamException(result);
+                                }
+
+                                if (bufferIndex == buffer.Length)
+                                {
+                                    bufferIndex = 0;
+                                }
+                            }
+                        }))
+                    {
+                        activity.RelatedError(metadata, $"{nameof(this.OnGetFileStream)}: TryCopyBlobContentStream failed");
+
+                        // TODO(Linux): Is this the correct Result to return?
+                        return Result.EFileNotFound;
+                    }
+                }
+                catch (GetFileStreamException e)
+                {
+                    return e.Result;
+                }
+
+                return Result.Success;
+            }
+            catch (Exception e)
+            {
+                EventMetadata metadata = this.CreateEventMetadata(relativePath, e);
+                metadata.Add(nameof(triggeringProcessId), triggeringProcessId);
+                metadata.Add(nameof(triggeringProcessName), triggeringProcessName);
+                metadata.Add(nameof(commandId), commandId);
+                this.LogUnhandledExceptionAndExit(nameof(this.OnGetFileStream), metadata);
+            }
+
+            return Result.EIOError;
+        }
+
+        private void OnLogError(string errorMessage)
+        {
+            this.Context.Tracer.RelatedError($"{nameof(LinuxFileSystemVirtualizer)}::{nameof(this.OnLogError)}: {errorMessage}");
+        }
+
+        private void OnFileModified(string relativePath)
+        {
+            try
+            {
+                if (Virtualization.FileSystemCallbacks.IsPathInsideDotGit(relativePath))
+                {
+                    this.OnDotGitFileOrFolderChanged(relativePath);
+                }
+            }
+            catch (Exception e)
+            {
+                this.LogUnhandledExceptionAndExit(nameof(this.OnFileModified), this.CreateEventMetadata(relativePath, e));
+            }
+        }
+
+        private Result NotifyFilePreConvertToFull(string relativePath)
+        {
+            this.OnFilePreConvertToFull(relativePath);
+            return Result.Success;
+        }
+
+        private Result OnPreDelete(string relativePath, bool isDirectory)
+        {
+            try
+            {
+                bool pathInsideDotGit = Virtualization.FileSystemCallbacks.IsPathInsideDotGit(relativePath);
+                if (pathInsideDotGit)
+                {
+                    if (relativePath.Equals(GVFSConstants.DotGit.Index, StringComparison.OrdinalIgnoreCase))
```

Comment:
This is a much larger question/discussion than this specific line (and no need to hold this PR up on the discussion).

Does VFS4G on Linux need to work with case-sensitive file systems?  If so, we'l need to find a way to make the comparison type part of the platform rather than having hard-coded `OrdinalIgnoreCase` (there will probably need to be a few other changes as well, but that's the big one that jumps out at me).

Reply:
Oooo ... good point.  Yes, that's going to be an issue; we don't want to match on `.GIT` or `.Git`, etc.  I think we'll have to take that off into another PR, for sure.

Reply:
Captured as github/VFSForGit#29.

Reply:
@chrisd8088 FYI you'll also need to adjust how the placeholder database is created (see #1140) as currently the table is always being created with a case-insensitive path/key and that might need to be OS dependent as well.

Reply:
(assuming you want to allow repos with file names that differ only in case)